src/components/menu/__internal__/submenu/submenu.component.js

```diff
@@ -273,6 +273,7 @@ const Submenu = React.forwardRef(
             tabIndex={asPassiveItem ? -1 : 0}
             variant={variant}
             inFullscreenView={inFullscreenView}
+            onClick={!asPassiveItem && !href ? () => {} : undefined}
```

Comment:
https://github.com/Sage/carbon/issues/5270 We've had this issue raised the other day, I wonder if we shouldn't add a default value of `() => {}` to the onClick prop and then change this to be like below as it would mean we can close the issue at the same time?

```
  onClick={!asPassiveItem && !href ? onClick : undefined}
```

Reply:
we cannot pass down the onClick method as it will be triggered twice - it is already handled in the component. The problem here is how the Link component is rendering a HTML element depending on the passed props.

Reply:
I take that back, I see that this is handled only in non fullscreen case. That could be simplified even more, as onClick even could be passed down without any conditionals.